src/View/Helper/FormHelper.php

```diff
@@ -558,6 +559,11 @@ public function secure(array $fields = [], array $secureAttributes = [])
         if (empty($this->request['_Token'])) {
             return null;
         }
+        $debugSecurity = Configure::read('debug');
+        if (isset($secureAttributes['debugSecurity'])) {
+            $debugSecurity = $secureAttributes['debugSecurity'];
+            unset($secureAttributes['debugSecurity']);
+        }
```

Comment:
we need to unset the secureAttributes, so I've fixed that to `$debugSecurity = $debugSecurity && $secureAttributes['debugSecurity'];`, did some cleanup and added extra unit tests :)